File: 141-linked-list-cycle/141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        if head == None or head.next == None: 
            return False
        
        myDict = {head: 0}
        point = head
        while point != None:
            point = point.next
            if myDict.get(point) == None:
                myDict[point] = 0
            else:
                return True
        
        return False

File: 141-linked-list-cycle/141-linked-list-cycle.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
bool hasCycle(struct ListNode *head) {
    struct ListNode *current = head;
    int index = 0;
    while (current) {
        struct ListNode *follow = head;
        int follow_index = 0;
        while (follow != current) {
            follow = follow->next;
            ++follow_index;
        }
        
        if (follow_index < index) {
            return true;
        }
        
        current = current->next;
        ++index;
    }
    return false;
}